### DISCUSSION TITLE
Better document behavior of comparison viewer dashboard with respect to term frequency adjustments?

### DISCUSSION THREAD

**Discussion by zmbc** [moj-analytical-services/splink#2353] [Ideas]
Based on some playing around, it looks like the comparison viewer dashboard essentially ignores term frequency adjustments, and the bar chart is of match weights pre-TF adjustments. Assuming this is intentional, it would be nice to write it down, since it tripped me up a bit.

**Comment by medwar99** [comment]
This confused me for a bit too, but it is intended behavior I believe as you get a sense for the base pre-TF adjusted behavior of a given comparison vector.

The bars in the chart represent a given permutation of the levels defined in your comparisons and so you can select a bar to see records that obtain that specific comparison vector. The pre-TF match weight of records in a given comparison vector are the same, but once the TF is applied the final match weight may be raised or lowered. 

e.g. John Smith <-> John Smith and  Willy Wonka <-> Willy Wonka will both meet "exact match" levels for both forename and surname, and so will be allocated the same comparison vector and be in the same bar on the visualization. However the TF for Willy Wonka may push the final match score higher (presuming Willy Wonka is an uncommon name of course). Selecting a threshold for the pre-TF adjusted value for this comparison vector would then allow you to keep edges where TF made them more likely even though they meet the same comparison levels. 

Agree that some documentation on how to effectively use it would be beneficial.

**Reply by zmbc** [reply]
Yep makes sense. I just found it confusing that when you click on a bar with e.g. probability 85.6%, _none_ of the pairs included in that bar need actually have that probability, if all the TF adjustments are non-zero.

**Comment by RobinL** [comment]
Thanks for the feedback.  I agree it's confusing - I've even confused myself with this in the past.

The reason it's never been fixed is that it's a bit tricky for two reasons:
- I'm not sure what the fix should be.  One option that doesn't feel perfect would be to attempt to pick example records which have small/zero tf adjustments.
- The dashboards are a bit fiddly/time consuming to update

Perhaps the issue could be mitigated by allowing term frequency adjustments to be shown/hidden (by default they could be in the 'hidden state'), with some caveat/warning displayed (although I generally prefer solutions that don't involve caveats because, speaking for myself, i often ignore/don't read them!)
